### PR TITLE
feat: compact mode in property pane

### LINKED ISSUE #5633: Set the row height from property pane control(compact view)
As a developer the row height options will not be available in the property pane rather than on the table component. The options should not be available to the end user

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Select compact Mode only in property pane

Fixes #5633 

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/set-row-height-from-property-pan 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.99 **(-0.01)** | 36.9 **(-0.01)** | 33.88 **(0.01)** | 55.54 **(-0.01)**
 :green_circle: | app/client/src/components/designSystems/appsmith/TableComponent/Table.tsx | 61.63 **(0)** | 38.89 **(0.71)** | 38.89 **(0)** | 60.98 **(0)**
 :red_circle: | app/client/src/components/designSystems/appsmith/TableComponent/TableHeader.tsx | 48.72 **(-1.28)** | 51.06 **(-0.94)** | 30 **(0)** | 45.95 **(-1.42)**
 :red_circle: | app/client/src/components/designSystems/appsmith/TableComponent/TableStyledWrappers.tsx | 67.82 **(0)** | 21.11 **(-0.74)** | 50 **(0)** | 67.82 **(0)**
 :green_circle: | app/client/src/components/designSystems/appsmith/TableComponent/index.tsx | 27.27 **(0)** | 2.63 **(0.07)** | 15.79 **(0)** | 25.68 **(0)**
 :red_circle: | app/client/src/utils/migrations/TableWidget.ts | 75.18 **(-0.17)** | 70.45 **(0)** | 83.33 **(0)** | 78.74 **(-0.17)**
 :green_circle: | app/client/src/widgets/TableWidget/index.tsx | 26.98 **(-0.16)** | 5.98 **(0.06)** | 25.49 **(0.49)** | 27.1 **(-0.17)**
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableActionIcon.tsx~~ | ~~77.78~~ | ~~50~~ | ~~50~~ | ~~77.78~~
 :x: | ~~app/client/src/components/designSystems/appsmith/TableComponent/TableCompactMode.tsx~~ | ~~72.73~~ | ~~41.67~~ | ~~25~~ | ~~70~~</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts [hunk 1]
```diff
@@ -848,6 +848,29 @@ export default [
...
       },
+      {
+        propertyName: "compactMode",
+        helpText: "Selects row height",
+        label: "Select Compact mode",
+        controlType: "DROP_DOWN",
+        defaultValue: "DEFAULT",
+        isBindProperty: true,
+        isTriggerProperty: false,
+        options: [
```
Comment (marks0351) [inline app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts:859]: should the values be all caps?

**Review by marks0351 — CHANGES_REQUESTED** [review]
I think we should use the word Row height  instead of compact mode in the property pane.
Also the toggle for row height visibility is still present in property pane.

cc: @somangshu @Tooluloope

File: app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts (hunk 1 above)
Comment (Tooluloope) [reply app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts:859]: @marks0351 , yeah that's the accepted value type.

I'd change it to Row height

Comment (marks0351) [reply app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts:859]: @Tooluloope I was asking about the labels also being all caps.

**Comment by marks0351** [comment]
@Tooluloope pls do add test cases(jest/cypress) as well.

File: app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts (hunk 1 above)
Comment (somangshu) [reply app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts:859]: If we are talking about the dropdown options then they should be First letter caps which is consistent to the rest of the widgets @marks0351 @Tooluloope

**Review by marks0351 — APPROVED** [review]

**Comment by somangshu** [comment]
@marks0351 can you take a call on if this needs testing and move it to the required pipeline further?

**Review by marks0351 — DISMISSED** [review]

**Comment by marks0351** [comment]
dp LGTM. we can merge this to release.

**Comment by somangshu** [comment]
Rerunning all jobs here, `Table_GeneralProperty_spec.js` failed

**Comment by somangshu** [comment]
Apologies, the failing test is related to the PR, Please check [here](https://dashboard.cypress.io/projects/eyxvp8/runs/9169/test-results/19ee81da-4394-4aed-ab94-127ee000fb5f?statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D), The test cannot find the row height property. @Tooluloope please fix this

**Comment by Tooluloope** [comment]
> Apologies, the failing test is related to the PR, Please check [here](https://dashboard.cypress.io/projects/eyxvp8/runs/9169/test-results/19ee81da-4394-4aed-ab94-127ee000fb5f?statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D), The test cannot find the row height property. @Tooluloope please fix this

Fixed it already.

**Review by jsartisan — DISMISSED** [review]

**Review by somangshu — DISMISSED** [review]

**Comment by somangshu** [comment]
@Tooluloope please clear out the merge conflict, get and approval and merge this ASAP

**Review by somangshu — DISMISSED** [review]

**Comment by somangshu** [comment]
@Tooluloope please bring the row-height property below the Default selected row

**Review by somangshu — APPROVED** [review]
LGTM

**Pull request merged**